core/src/conn.rs

```diff
@@ -234,8 +234,9 @@ impl MongoConnection {
                 // Ensure the library is loaded if Enterprise edition is detected
                 if get_mongosqltranslate_library().is_none() {
                     return Err(Error::UnsupportedClusterConfiguration(
-                        "Enterprise edition detected, but mongosqltranslate library not found."
-                            .to_string(),
+                        "Enterprise edition detected, but mongosqltranslate library not found. \
+                        You are either (1) trying to connect to an on-premises cluster and have the wrong driver \
+                        or are (2) trying to connect to an Atlas cluster directly instead of through an ADF instance.".to_string(),
```

Comment:
```suggestion
                        or are (2) trying to connect directly to an Atlas cluster directly instead of through a Data Federation instance.".to_string(),
```